modules/aardvarkBidAdapter.js

```diff
@@ -0,0 +1,158 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'aardvark';
+const DEFAULT_ENDPOINT = 'bidder.rtk.io';
+const SYNC_ENDPOINT = 'sync.rtk.io';
+const AARDVARK_TTL = 300;
+const AARDVARK_CURRENCY = 'USD';
+
+let hasSynced = false;
+
+export function resetUserSync() {
+  hasSynced = false;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+
+  isBidRequestValid: function(bid) {
+    return ((typeof bid.params.ai === 'string') && !!bid.params.ai.length &&
+        (typeof bid.params.sc === 'string') && !!bid.params.sc.length);
+  },
+
+  buildRequests: function(validBidRequests, bidderRequest) {
+    var auctionCodes = [];
+    var requests = [];
+    var requestsMap = {};
+    var referer = utils.getTopWindowUrl();
+    var pageCategories = [];
+
+    if (window.top.rtkcategories && Array.isArray(window.top.rtkcategories)) {
+      pageCategories = window.top.rtkcategories;
+    }
+
+    utils._each(validBidRequests, function(b) {
+      var rMap = requestsMap[b.params.ai];
+      if (!rMap) {
+        rMap = {
+          shortCodes: [],
+          payload: {
+            version: 1,
+            jsonp: false,
+            rtkreferer: referer
+          },
+          endpoint: DEFAULT_ENDPOINT
+        };
+
+        if (pageCategories && pageCategories.length) {
+          rMap.payload.categories = pageCategories.slice(0);
+        }
+
+        if (b.params.categories && b.params.categories.length) {
+          rMap.payload.categories = rMap.payload.categories || []
+          utils._each(b.params.categories, function(cat) {
+            rMap.payload.categories.push(cat);
+          });
+        }
+
+        if (bidderRequest && bidderRequest.gdprConsent) {
+          rMap.payload.gdpr = false;
+          if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+            rMap.payload.gdpr = bidderRequest.gdprConsent.gdprApplies;
+          }
+          if (rMap.payload.gdpr) {
+            rMap.payload.consent = bidderRequest.gdprConsent.consentString;
+          }
+        }
+
+        requestsMap[b.params.ai] = rMap;
+        auctionCodes.push(b.params.ai);
+      }
+
+      rMap.shortCodes.push(b.params.sc);
+      rMap.payload[b.params.sc] = b.bidId;
+
+      if ((typeof b.params.host === 'string') && b.params.host.length &&
+          (b.params.host !== rMap.endpoint)) {
+        rMap.endpoint = b.params.host;
+      }
+    });
+
+    utils._each(auctionCodes, function(auctionId) {
+      var req = requestsMap[auctionId];
+      requests.push({
+        method: 'GET',
+        url: `//${req.endpoint}/${auctionId}/${req.shortCodes.join('_')}/aardvark`,
+        data: req.payload,
+        bidderRequest
+      });
+    });
+
+    return requests;
+  },
+
+  interpretResponse: function(serverResponse, bidRequest) {
+    var bidResponses = [];
+    utils._each(serverResponse.body, function(rawBid) {
```

Comment:
Can you review the logic in using this `utils._each` function?  It's currently iterating through each of the properties of the response object and evaluating them each as an entire bid object.  

This is a copy of the bid response object that I saw when I tested and it treated `rawBid` as `demo_exchange` (from  the `ex` field) during the first pass of the function.
```
{"ex": "demo_exchange", "width": 300, "cpm": 0.01, "cid": "2edd6ea6821b56", "media": "banner", "adm": "<div>Demo Unit</div>", "id": "1234", "height": 250}
```

Reply:
We've been updating the demo auction today, sorry about that. That is an array returned from our endpoint, so it gets iterated over as an array, not a map. It is now an array as it should be.